src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -165,6 +164,27 @@ protected override void InitializeWorker(CompilationStartAnalysisContext context
 
             var guidParseMethods = guidType?.GetMembers("Parse") ?? ImmutableArray<ISymbol>.Empty;
 
+            var convertType = typeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemConvert);
+            var superfluousConvertToStringFormatProviderOverloads = convertType?.GetMembers(nameof(Convert.ToString))
+                .OfType<IMethodSymbol>()
+                .Where(m => m.IsStatic
+                            && m.Parameters is [{ Type.SpecialType: SpecialType.System_String or SpecialType.System_Boolean or SpecialType.System_Char }, var possibleFormatProvider]
+                            && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)) ?? [];
+            var superfluousConvertToToCharFormatProviderOverloads = convertType?.GetMembers(nameof(Convert.ToChar))
+                .OfType<IMethodSymbol>()
+                .Where(m => m.IsStatic
+                            && m.Parameters is [{ Type.SpecialType: SpecialType.System_String }, var possibleFormatProvider]
+                            && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)) ?? [];
+            var superfluousConvertToBooleanFormatProviderOverloads = convertType?.GetMembers(nameof(Convert.ToBoolean))
+                .OfType<IMethodSymbol>()
+                .Where(m => m.IsStatic
+                            && m.Parameters is [{ Type.SpecialType: SpecialType.System_String }, var possibleFormatProvider]
+                            && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)) ?? [];
+            ImmutableHashSet<IMethodSymbol> superfluousFormatProviderOverloads = superfluousConvertToStringFormatProviderOverloads
+                .Concat(superfluousConvertToToCharFormatProviderOverloads)
+                .Concat(superfluousConvertToBooleanFormatProviderOverloads)
+                .ToImmutableHashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
```

Comment:
Got some NITs:
- Better to check null before head to avoid null check each reference and creating empty array
- For ToChar, ToBoolean we expect only one method to get, no need to create array for this 
- `m.IsStatic` is redundant as Convert is a static type, all methods are static anyways
Something like:
```suggestion
            ImmutableHashSet<IMethodSymbol> superfluousFormatProviderOverloads = ImmutableHashSet<IMethodSymbol>.Empty;
            if (convertType != null)
            {
                superfluousFormatProviderOverloads = convertType.GetMembers(nameof(Convert.ToString))
                    .OfType<IMethodSymbol>()
                    .Where(m => m.Parameters is [{ Type.SpecialType: SpecialType.System_String or SpecialType.System_Boolean or SpecialType.System_Char }, var possibleFormatProvider]
                                && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)).ToImmutableHashSet();
                superfluousFormatProviderOverloads = superfluousFormatProviderOverloads
                    .Add(convertType.GetMembers(nameof(Convert.ToChar))
                        .OfType<IMethodSymbol>()
                        .First(m => m.Parameters is [{ Type.SpecialType: SpecialType.System_String }, var possibleFormatProvider]
                                    && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)))
                    .Add(convertType.GetMembers(nameof(Convert.ToBoolean))
                        .OfType<IMethodSymbol>()
                        .First(m => m.Parameters is [{ Type.SpecialType: SpecialType.System_String }, var possibleFormatProvider]
                                    && possibleFormatProvider.Type.Equals(iformatProviderType, SymbolEqualityComparer.Default)));
            }
```

Reply:
Thanks for the review!